torch/_inductor/codegen/cpp.py

```diff
@@ -4901,6 +4910,10 @@ def template_buffer_has_other_users(
         with V.set_kernel_handler(kernel):
             node_schedule = [template_node, *epilogue_nodes]
             kernel_name = self.define_kernel(src_code, node_schedule, kernel.args)
+        for line in grouped_gemm_allocate_lines:
+            if kernel.args.output_buffers[line.node.get_name()] != "REMOVED":
+                # only allocate the buffer when it's not removed in kernel args
+                V.graph.wrapper_code.writeline(line)
```

Comment:
similar comment in the base pr, if you do this in the scheduler, there is existing logic for checking it.

Reply:
Thanks for the comment and changed to use `node.mark_run` which will check if the buffer has been removed or not.